radicle-cli/src/git.rs

```diff
@@ -175,24 +175,35 @@ pub fn is_signing_configured(repo: &Path) -> Result<bool, anyhow::Error> {
 }
 
 /// Return the list of radicle remotes for the given repository.
-pub fn remotes(repo: &git2::Repository) -> anyhow::Result<Vec<(String, NodeId)>> {
-    let mut remotes = Vec::new();
-
-    for name in repo.remotes().iter().flatten().flatten() {
-        let remote = repo.find_remote(name)?;
-        for refspec in remote.refspecs() {
-            if refspec.direction() != git2::Direction::Fetch {
-                continue;
-            }
-            if let Some((peer, _)) = refspec.src().and_then(self::parse_remote) {
-                remotes.push((name.to_owned(), peer));
-            }
-        }
-    }
-
+pub fn rad_remotes(repo: &git2::Repository) -> anyhow::Result<Vec<git2::Remote>> {
+    let remotes: Vec<_> = repo
+        .remotes()?
+        .iter()
+        .filter_map(|name| {
+            name.and_then(|name| repo.find_remote(name).ok())
+                .and_then(|remote| {
+                    let Some(url) = remote.url() else {
+                        return None
+                    };
+
+                    if url.starts_with(radicle::storage::git::transport::local::Url::SCHEME) {
+                        Some(remote)
+                    } else {
+                        None
+                    }
+                })
```

Comment:
These let/if/else blocks can be further collapsed using `and_then`

Reply:
I _think_ you can use `?` because you're in a block that returns an `Option`.

Reply:
@cloudhead  How I can return the `git2::Remote` with `and_then` if I need to check before the URL?

I can call `and_then` returning the remote if the URL is some, but I do not see a difference with the actual implementation? just another way to write the same thing? we are not removing the conditional check, or I am missing something?

@FintanH Mh, can you add the suggestion of what you mean? in 10 lines of code, there are too many places where can add `?` and return an `Option` 😸

Reply:
Sure, you can do this:
```
         .remotes()?
         .iter()
         .filter_map(|name| {
-            name.and_then(|name| repo.find_remote(name).ok())
-                .and_then(|remote| {
-                    let Some(url) = remote.url() else {
-                        return None
-                    };
-
-                    if url.starts_with(radicle::storage::git::transport::local::Url::SCHEME) {
-                        Some(remote)
-                    } else {
-                        None
-                    }
-                })
+            let name = name?;
+            let remote = repo.find_remote(name).ok()?;
+            let url = remote.url()?;
+            url.starts_with(radicle::storage::git::transport::local::Url::SCHEME)
+                .then_some(remote)
         })
         .collect();
     Ok(remotes)
```

Reply:
Ah yeah I like your approach also if I tend to avoid using to many `?` because it will make hard the code to debug, but I think in this case there is no difference, so let try this